Makefile

```diff
@@ -380,6 +380,10 @@ build-gadgets: install/ig
 push-gadgets: install/ig
 	$(MAKE) -C gadgets/ push
 
+.PHONY: test-gadgets
+test-gadgets: install/ig
+	$(MAKE) -C gadgets/ test -o build
```

Comment:
If you're not building anymore the gadgets, then whoever execute `test-gadgets` needs to ensure the gadgets were already built and pushed. Meaning that the `test-gadgets` job has to be in level 3 and have `build-and-push-gadgets` in its `needs`.

Reply:
Right, I am going to revert this change and directly use the test command, thanks for pointing out the level 3 detail.

Reply:
Mmmm this change is fine, just need to update the CI job dependecies.

Reply:
I'm not totally sure this rule should skip building the gadgets. `make tests-gadgets` should do all the work needed to run the tests (even building the gadgets). If we want to avoid rebuilding them in the CI, we can use something like `make -C gadgets/ test -o build` directly from the action workflow.

Reply:
CI Run: https://github.com/pawarpranav83/inspektor-gadget/actions/runs/8640186743/job/23687887510.
The test is passing locally, any reason why its giving unknown flag error only in CI?

Reply:
mmm, I think it's because the timeout parameter is temporary gone as stated in https://github.com/inspektor-gadget/inspektor-gadget/issues/2653. @flyth do you know how difficult it's to add that parameter back?

Reply:
Should be fairly easy, can take a look at it later today.

Reply:
@pawarpranav83 perhaps you can switch to StartAndStop as it was before until we add the timeout parameter back?